src/main/java/com/philips/research/spdxbuilder/persistence/ort/OrtReader.java

```diff
@@ -30,137 +30,137 @@
-            p.scopes.forEach(scope -> {
-                System.out.println("+ Adding scope '" + scope.name + "'");
-                scope.putAllDependencies(dictionary);
-            });
-        });
-    }
-
-    private void registerPackages(ResultJson result, BillOfMaterials bom, HashMap<String, Package> dictionary) {
-        result.packages()
-                .filter(pkg -> dictionary.containsKey(pkg.id))
-                .forEach(pkg -> {
-                    if (pkg.id == null) {
-                        return;
-                    }
-                    final var created = pkg.createPackage();
-                    dictionary.put(pkg.id, created);
-                    bom.addPackage(created);
-                });
-    }
-
-    private void registerRelations(ResultJson result, BillOfMaterials bom, HashMap<String, Package> dictionary) {
-        result.projects.forEach(p -> {
-            final var project = dictionary.get(p.id);
-            p.scopes.stream()
-                    .flatMap(scope -> scope.dependencies.stream())
-                    .forEach(dep -> dep.registerRelations(bom, project, dictionary));
-        });
-    }
+  private static final FileSystem FILE_SYSTEM = FileSystems.getDefault();
+  private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory())
+	.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+	.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.NON_PRIVATE)
+	.setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE);
+
+  private final File file;
+  private final Map<String, @NullOr URI> projectPackages = new HashMap<>();
+  private final Map<String, List<String>> projectExcludes = new HashMap<>();
+
+  public OrtReader(File file) {
+	this.file = file;
+  }
+
+  /**
+   * Marks a project for import with an (optional) package alias
+   *
+   * @param id   ORT identifier for the project
+   * @param purl (Optional) Package URL
+   */
+  public OrtReader defineProjectPackage(String id, @NullOr URI purl) {
+	//noinspection ConstantConditions
+	projectPackages.put(id, purl);
+	return this;
+  }
+
+  /**
+   * Adds excluded scopes to ORT.
+   *
+   * @param id       ORT identifier for the project
+   * @param excluded (optionally wild-carded) scope names
+   */
+  public OrtReader excludeScopes(String id, List<String> excluded) {
+	projectExcludes.put(id, excluded);
+	return this;
+  }
+
+  @Override
+  public void read(BillOfMaterials bom) {
+	System.out.println("Reading ORT file '" + file + "'...");
+	try {
+	  final var yaml = MAPPER.readValue(file, OrtJson.class);
+	  final var dictionary = new HashMap<String, Package>();
+
+	  if (yaml.analyzer == null || yaml.analyzer.result == null) {
+		throw new OrtReaderException("ORT file does not include an 'analyzer.result' section");
+	  } else if (yaml.analyzer.result.hasIssues) {
+		throw new OrtReaderException("The analyzed ORT file has issues, unable to generate a valid SPDX file");
```

Comment:
I don't think the "file has issues"; the flag is set if the ORT Analyzer finished with issues.

A valid SPDX file can still be produced, but it might not contain all information; I would not mind if we completely drop the text behind the comma.